experiments/train_se3_diffusion.py

```diff
@@ -228,21 +260,51 @@ def start_training(self, return_logs=False):
             replica_id = 0
         if self._use_wandb and replica_id == 0:
                 self.init_wandb()
+        assert(not self._exp_conf.use_ddp or self._exp_conf.use_gpu)
 
+        # GPU mode
         if torch.cuda.is_available() and self._exp_conf.use_gpu:
-            gpu_id = self._available_gpus[replica_id]
-            device = f"cuda:{gpu_id}"
+            # single GPU mode
+            if self._exp_conf.num_gpus==1 :
+                gpu_id = self._available_gpus[replica_id]
+                device = f"cuda:{gpu_id}"
+                self._model = self.model.to(device)
+                self._log.info(f"Using device: {device}")
+            #muti gpu mode
+            elif self._exp_conf.num_gpus > 1:
+                device_ids = [
+                f"cuda:{i}" for i in self._available_gpus[:self._exp_conf.num_gpus]
+                ]
+                #DDP mode
+                if self._use_ddp :
+                    device = torch.device("cuda",self.ddp_info['local_rank'])
+                    model = self.model.to(device)
+                    self._model = DDP(model, device_ids=[self.ddp_info['local_rank']], output_device=self.ddp_info['local_rank'],find_unused_parameters=True)
+                    self._log.info(f"Multi-GPU training on GPUs in DDP mode, node_id : {self.ddp_info['node_id']}, devices: {device_ids}")
+                #DP mode
+                else:
+                    assert(len(self._available_gpus)>self._exp_conf.num_gpus,f"require {self._exp_conf.num_gpus} GPUs, but only {len(self._available_gpus)} GPUs available ")
+                    self._model = self.model.to(device)
+                    self._model = DP(self._model, device_ids=device_ids)
+                    self._log.info(f"Multi-GPU training on GPUs in DP mode: {device_ids}")
```

Comment:
Can you put a final else with a message along the lines of: self._info.info("GPU requested but not being used!")